repose-aggregator/commons/utilities/src/main/java/org/openrepose/commons/utils/servlet/http/MutableHttpServletResponse.java

```diff
@@ -302,11 +302,16 @@ public void sendError(int code) throws IOException {
 
     @Override
     public void sendError(int code, String message) {
-        super.setStatus(code);
-        this.message = message;
+        setStatus(code, message);
         error = true;
     }
 
+    @Override
+    public void setStatus(int code, String message) {
+        super.setStatus(code, message);
+        this.message = message;
```

Comment:
Another way to do this would be to have `DerpFilter` add a *Delegation Message* header. This would propagate through all the wrappers. The `ResponseMessagingService` could retrieve it and removed it before being returned to the client.